llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -1847,9 +1847,13 @@ void NVPTXAsmPrinter::bufferLEByte(const Constant *CPV, int Bytes,
   auto AddIntToBuffer = [AggBuffer, Bytes](const APInt &Val) {
     size_t NumBytes = (Val.getBitWidth() + 7) / 8;
     SmallVector<unsigned char, 16> Buf(NumBytes);
-    for (unsigned I = 0; I < NumBytes; ++I) {
+    for (unsigned I = 0; I < NumBytes - 1; ++I) {
```

Comment:
This could certainly use a comment or two about what's going on, because it's not obvious. At all.

Why exactly are we special casing the last byte?

Reply:
It's explained in the PR description, also look at the test. If the type's bitwidth is not a multiple of 8, the old code crashes.

Reply:
PR description only shows the expression for the failed assertion: `bitPosition < BitWidth && (numBits + bitPosition) <= BitWidth`, but it was not clear which part of that assertion has failed.

The substance of the fix LGTM.

That said, If I ever need to look at this code in the future, I'm pretty sure I'll have to scratch my head, again, pondering what's going on here. Hence, my suggestion that a comment would be helpful. We already have way too many puzzles, we do not need more.

Reply:
Sounds good. Sent a follow-up.